packages/plugin-ext/src/main/browser/plugin-ext-frontend-module.ts

```diff
@@ -60,9 +60,16 @@ import { ViewColumnService } from './view-column-service';
 import { ViewContextKeyService } from './view/view-context-key-service';
 import { PluginViewWidget, PluginViewWidgetIdentifier } from './view/plugin-view-widget';
 import { TreeViewWidgetIdentifier, VIEW_ITEM_CONTEXT_MENU, PluginTree, TreeViewWidget, PluginTreeModel } from './view/tree-view-widget';
+import { RPCProtocol } from '../../common/rpc-protocol';
+import { LanguagesMainFactory, OutputChannelRegistryFactory } from '../../common';
+import { LanguagesMainImpl } from './languages-main';
+import { OutputChannelRegistryMainImpl } from './output-channel-registry-main';
 
 export default new ContainerModule((bind, unbind, isBound, rebind) => {
 
+    bind(LanguagesMainFactory).toFactory(context => (proxy: RPCProtocol) => new LanguagesMainImpl(proxy, context.container));
```

Comment:
I hope to put everything in DI context, i.e.
```ts
    bind(LanguagesMainImpl).toSelf().inTransientScope();
    bind(LanguagesMainFactory).toFactory(context => (rpc: RPCProtocol) => {
        const child = context.container.createChild();
        child.bind(RPCProtocol).toConstantValue(rpc);
        return child.get(LanguagesMainImpl);
    });
```

so `LanguagesMainImpl` can inject services and be redound by an extended extension, i.e.
```ts
rebind(LanguagesMainImpl).to(MyLanguagesMainImpl);
```

Reply:
One thing that is not clear to me is how to do wrapping using DI: what I mean is that we want to wrap whatever is the existing implementation. As I understand it, when you rebind, you need to provide a concrete implementation, not based on what is preexisting.

Reply:
@JPinkney provided an example here: https://github.com/JPinkney/theia-plugin-extension-metrics/blob/master/src/browser/plugin-metrics-languages-main.ts#L33